### ISSUE TITLE
Accessing db object in react router loaders and actions

### ISSUE THREAD

**Issue by kdabir** [kikko-land/kikko#157]
I am trying to fire queries in react router [loaders](https://reactrouter.com/en/main/route/loader) / [actions](https://reactrouter.com/en/main/route/action). Since loader/actions functions are neither react hooks, nor they are components, I cannot call hooks like `useDbQuery` in these functions. I get this error if I do:

<img width="554" alt="image" src="https://user-images.githubusercontent.com/735240/227511544-28df98b3-04b6-415f-9890-42d54d5fb350.png">

I cannot access to the db object as that created/managed by the `DbsHolder` and is only available inside hooks. I only pass the config at the instantiation 

```
<DbsHolder defaultDbConfig={config}>
    <EnsureDbLoaded fallback={<span>Loading db...</span>}>
...
```

So, to fix this: 

1. either DbsHolder accepts a db object, that I can create and pass to it
2. or, provide some other way to access the db object in loaders/action.

**Comment by quolpr** [comment]
@kdabir yep, you are right, we should have the way to access `dbs` without `DbsHolder`. The idea of

> either DbsHolder accepts a db object, that I can create and pass to it

Looks preferable to me, but then also you will need somewhere to add the code to close db on page leave (if necessary). Currently, `DbsHolder` has `useEffect` which stops db on demount.

Not sure about deadlines for now though
